### PR TITLE
New update_dockerfile_packages()

### PR DESCRIPTION
**Pull request by maurolepore**

### CODE REVIEW

File: bin/start-containter-with [hunk 1]
```diff
@@ -12,5 +12,5 @@ docker run --rm -ti \
   --mount type=bind,source="$data",target=/pacta-data \
   --mount type=bind,source="$report",target=/create_interactive_report \
   --mount type=bind,source="$stresstest",target=/StressTestingModelDev \
-  2diirunner:1.3.0 "$1"
+  2dii/2diidockerrunner "$1"
```
Comment (maurolepore) [inline bin/start-containter-with:15]: Use image from 2dii's dockerhub

File: R/utils.R [hunk 2]
```diff
@@ -29,3 +29,91 @@ resolve_conflicts <- function() {
...
+
+#' @examples
+#' some_dockerfile <- tempfile()
+#' writeLines(create_empty_dockerfile("library(dplyr)"), some_dockerfile)
+#'
+#' # Before
+#' writeLines(readLines(some_dockerfile))
+#' update_dockerfile_packages(path = some_dockerfile)
+#' # After
+#' writeLines(readLines(some_dockerfile))
```
Comment (maurolepore) [inline R/utils.R:41]: Here is how this works:

``` r
  
devtools::load_all()
#> Loading PACTA.analysis
some_dockerfile <- tempfile()
writeLines(create_empty_dockerfile("library(dplyr)"), some_dockerfile)

# Before
writeLines(readLines(some_dockerfile))
#> # update-dockerfile-packages-start
#> library(dplyr)
#> # update-dockerfile-packages-end
update_dockerfile_packages(path = some_dockerfile)
# After
writeLines(readLines(some_dockerfile))
#> # update-dockerfile-packages-start
#>     && Rscript -e "install.packages( \
#>              c( \
#>                'assertthat', \
#>                'bookdown', \
#>                'config', \
#>                'conflicted', \
#>                'countrycode', \
#>                'data.table', \
#>                'devtools', \
#>                'dplyr', \
#>                'fs', \
#>                'fst', \
#>                'ggplot2', \
#>                'glue', \
#>                'here', \
#>                'janitor', \
#>                'jsonlite', \
#>                'knitr', \
#>                'purrr', \
#>                'readr', \
#>                'readxl', \
#>                'renv', \
#>                'reshape2', \
#>                'rlang', \
#>                'rmarkdown', \
#>                'rstudioapi', \
#>                'scales', \
#>                'stringr', \
#>                'testthat', \
#>                'tibble', \
#>                'tidyr', \
#>                'tidyselect', \
#>                'usethis', \
#>                'withr', \
#>                'writexl', \
#>                'zoo' \
#>              ) \
#>            )" \
#> # update-dockerfile-packages-end
```

<sup>Created on 2020-10-13 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

File: deduplicate/load-and-attach-r-packages.R [hunk 3]
```diff
@@ -1,36 +1,34 @@
-suppressPackageStartupMessages({
-  library(assertthat)
```
Comment (maurolepore) [inline deduplicate/load-and-attach-r-packages.R:2]: I moved this to inside `use_r_packages()`

File: R/utils.R [hunk 4]
```diff
@@ -13,8 +13,8 @@ setup_project <- function() {
   source(path)
 }
 
-use_r_packages <- function(path = "deduplicate/load-and-attach-r-packages.R") {
-  source(path)
+use_r_packages <- function(path = packages_path()) {
+  suppressPackageStartupMessages(source(path))
```
Comment (maurolepore) [inline R/utils.R:17]: I moved this here.

**Review by maurolepore — COMMENTED** [review]
Here is my updated work about docker. The work on the Dockerfiles belongs to a different repo: 2diidockerrunner.

File: R/utils.R (hunk 4 above)
Comment (cjyetman) [reply R/utils.R:17]: probably much more sensible here 👍

File: deduplicate/load-and-attach-r-packages.R (hunk 3 above)
Comment (cjyetman) [reply deduplicate/load-and-attach-r-packages.R:2]: 👍

**Pull request merged**